optuna/visualization/_contour.py

```diff
@@ -34,31 +31,31 @@
 
 class _AxisInfo(NamedTuple):
     name: str
-    range: Tuple[float, float]
+    range: tuple[float, float]
     is_log: bool
     is_cat: bool
-    indices: List[Union[str, int, float]]
-    values: List[Union[str, float, None]]
+    indices: list[str | int | float]
+    values: list[str | float | None]
 
 
 class _SubContourInfo(NamedTuple):
     xaxis: _AxisInfo
     yaxis: _AxisInfo
-    z_values: Dict[Tuple[int, int], float]
+    z_values: dict[tuple[int, int], float]
 
 
 class _ContourInfo(NamedTuple):
-    sorted_params: List[str]
-    sub_plot_infos: List[List[_SubContourInfo]]
+    sorted_params: list[str]
+    sub_plot_infos: list[list[_SubContourInfo]]
     reverse_scale: bool
     target_name: str
 
 
 def plot_contour(
     study: Study,
-    params: Optional[List[str]] = None,
+    params: list[str] | None = None,
     *,
-    target: Optional[Callable[[FrozenTrial], float]] = None,
+    target: Callable[[FrozenTrial], float] | None = None,
```

Comment:
[question] Though `typing.Optional` seems not to be deprecated, should we replace `typing.Optional` with `| None`? Or is it just a matter of preference?

Reply:
It's just a matter of preference, but I think `X | None` is simpler than `Optional[X]`.